internal/emscripten/emscripten.go

```diff
@@ -82,15 +104,69 @@ func (v *InvokeFunc) Call(ctx context.Context, mod api.Module, stack []uint64) {
 	tableOffset := wasm.Index(stack[0]) // position in the module's only table.
 	copy(stack, stack[1:])              // pop the tableOffset.
 
+	// The Go implementation below mimics the Emscripten JS behaviour to support
+	// longjmps from indirect function calls. The implementation of these
+	// indirection function calls in Emscripten JS is like this:
+	//
+	// function invoke_iii(index,a1,a2) {
+	//  var sp = stackSave();
+	//  try {
+	//    return getWasmTableEntry(index)(a1,a2);
+	//  } catch(e) {
+	//    stackRestore(sp);
+	//    if (e !== e+0) throw e;
+	//    _setThrew(1, 0);
+	//  }
+	//}
+
+	// This is the equivalent of "var sp = stackSave();".
+	// We reuse savedStack to save allocations. We allocate with a size of 2
+	// here to accommodate for the input and output of setThrew.
+	var savedStack [2]uint64
+	err = mod.ExportedFunction("stackSave").CallWithStack(ctx, savedStack[:])
+	if err != nil {
+		panic(err)
+	}
+
 	// Lookup the table index we will call.
 	t := m.Tables[0] // Note: Emscripten doesn't use multiple tables
 	f, err := m.Engine.LookupFunction(t, typeID, tableOffset)
 	if err != nil {
 		panic(err)
 	}
-
 	err = f.CallWithStack(ctx, stack)
+	runtime.KeepAlive(t)
 	if err != nil {
```

Comment:
I think this block hasn't been covered in the tests. Do you want to add it?

Reply:
@mathetake Sure, but I'm not that great in writing wat files, I can convert the sample C into a test and add the Emscripten compiled one to the test folder? Would that suffice?

Reply:
I would prefer Wat, so it would be great if you could do it. If it's too difficult, @evacchi or I would take over so feel free to tell us!